### ISSUE TITLE
Unable to import scvelo as scv

### ISSUE THREAD

**Issue by imilenkovic** [theislab/cellrank#544]
After installing cellrank with conda and creating the cellrank environment, I activate the environment, and in python I try to import scvelo but I can't import it at all.

```python
Python 3.9.2 | packaged by conda-forge | (default, Feb 21 2021, 05:02:46)
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import scvelo as scv
Illegal instruction (core dumped)
$

```

```pytb
Python 3.9.2 | packaged by conda-forge | (default, Feb 21 2021, 05:02:46)
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import scvelo as scv
Illegal instruction (core dumped)
$

```

#### Versions:

> ...

**Comment by michalk8** [comment]
Hi @imilenkovic ,

based on what I can see, it looks like some packages have been installed incorrectly. scVelo does not depend on CellRank (it's vice-versa), so I'd assume one of it's dependencies, such as numpy/umap-learn/etc. could be broken in your environment.
I also can't reproduce this on 3.9 using conda to install cellrank (unsatisfiable requirements on my system), with normal `pip` it's fine.
I'd try creating a fresh new environment with a lower Python version (3.8) and reinstalling the packages. If it still persists, I'd try to determine which package causes the core to be dumped by importing them 1 by 1.

**Comment by imilenkovic** [comment]
Hello @michalk8,

Thanks a lot! Using python=3.8 and installing cellrank using pip worked well!

I tried following the guidelines, and after importing the libraries this happened:

>>> scv.settings.set_figure_params('scvelo')
In :

whatever I tried after this "In : " nothing would happen, did I do something wrong here?

I tried to skip this line and went on with the commands; after running "sc.pp.neighbors(adata, n_pcs=30, n_neighbors=30)" I got a long error finshing with:

numba.core.errors.LoweringError: Failed in nopython mode pipeline (step: nopython mode backend)
Storing i64 to ptr of i32 ('dim'). FE type int32

`File "../../../../users/enovoa/imilenkovic/software/miniconda3/envs/cellrank/lib/python3.8/site-packages/umap/layouts.py", line 52:
def rdist(x, y):
    <source elided>
    result = 0.0
    dim = x.shape[0]
    ^
`

Do you maybe know why this happened? Before running cellrank, I ran velocyto to generate the loom file, and then Seurat to convert it h5ad. Did something perhaps fail in those previous steps? And is h5ad the only format I should work with in cellrank? Thanks a lot!

**Comment by michalk8** [comment]
> whatever I tried after this "In : " nothing would happen, did I do something wrong here?

I'm not sure if I understand correctly - you ran `scv.settings.set_figure_params('scvelo')` and nothing happened in the notebook (i.e. no output)? This just sets plotting parameters, so it shouldn't return/print anything.

The error you're receiving now is also being mentioned in the scanpy issue: https://github.com/theislab/scanpy/issues/1756
I've looked at the `rdist` numba code, it thinks `x.shape[0]` is of type `int64` and tries to store it in `int32`, which would result in a loss of precision.
I am using `umap-learn=='0.5.1' numba=='0.52.0' numpy='1.20.1' llvmlite=='0.35.0'` and can't reproduce this on some data that I have (also tried it with `numba=='0.53.0'`, still worked).
First, I'd try running it with the above versions. If that does not help, it can be that during those conversion processes, some datatypes may have been converted incorrectly (not sure how it could happen for shapes though).

**Comment by imilenkovic** [comment]
I am running the commands in the command line:

Python 3.8.8 | packaged by conda-forge | (default, Feb 20 2021, 16:22:27)
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import scvelo as scv
>>> import scanpy as sc
>>> import cellrank as cr
>>> import numpy as np
>>> scv.settings.verbosity = 3
>>> scv.settings.set_figure_params('scvelo')
In :

So after running the last line I only get this "In : ".
Do I need to run this in a jupyter notebook or command line should also be fine?

I will try using the versions you suggest to overcome the other error.
Thank you!

**Comment by michalk8** [comment]
> So after running the last line I only get this "In : ".

Ah, I see. No this seems to be just some artifact from jupyter notebooks, nothing else - you don't need a notebook to run this.

**Comment by imilenkovic** [comment]
> > So after running the last line I only get this "In : ".
> 
> Ah, I see. No this seems to be just some artifact from jupyter notebooks, nothing else - you don't need a notebook to run this.

But is there a way to overcome it in the command line? Because the only thing I can do to get out of it is ctrl+D and it shuts off python completely

**Comment by michalk8** [comment]
@imilenkovic now I see what you mean. To reproduce:
```python3
import scvelo as scv
scv.settings.set_figure_params('scvelo')
```
also puts me inside what looks like a different python interpreter (because `scv` there is not available). This is definitely a bug and you should create an issue on https://github.com/theislab/scvelo/issues (with `scanpy`, this doesn't happen).
For now, you can either:
1. skip this line, since it ony sets figure parameters
2. launch Python as `ipython` instead of `python` - there it works without plunging you into the new interpreter

**Comment by millersan** [comment]
finish with numba=='0.52.0', fail in numba=='0.53.1'

**Comment by Marius1311** [comment]
Hi @millersan, are you saying you're experiencing a bug in more recent numba versions?

**Comment by millersan** [comment]
> Hi @millersan, are you saying you're experiencing a bug in more recent numba versions?
Yes, when I install numba=='0.52.0', the pipeline can run.

**Comment by millersan** [comment]
> Hi @millersan, are you saying you're experiencing a bug in more recent numba versions?

Yes, when I install numba=='0.52.0', the pipeline can run.

**Comment by Marius1311** [comment]
@michalk8, can you take over please?

**Comment by michalk8** [comment]
> Yes, when I install numba=='0.52.0', the pipeline can run.

numba is pinned in https://github.com/theislab/cellrank/blob/master/requirements.txt#L9 to `>=0.52.2,<0.53.0`

**Comment by Marius1311** [comment]
Okay, so everything is fine here.

**Comment by mihem** [comment]
Sorry, encounter the same problem with numba == 0.52.0.
In a fresh virtual environment using python 3.8.9 and scvelo develop (because not develop version with umap-learn<0.5 runs into problems with moments), no problem running `scv.settings.set_figure_params(style=None)` . After installing cellrank, I am stuck in "In" after `scv.settings.set_figure_params('scvelo')` as described above.

**Comment by michalk8** [comment]
@mihem 

The problem seems to be this line in scvelo: https://github.com/theislab/scvelo/blob/master/scvelo/settings.py#L300
The quick workaround is to use:
```python
scv.settings.set_figure_params('scvelo', ipython_format=None)
```
this will cause an Exception to be raised which is caught immediatelly afterwards. I've created an issue on `scvelo` https://github.com/theislab/scvelo/issues/437.
The reason why you're getting this after installing CellRank is that in our requirements, we have `ipywidgets`, which also pulls the `ipython`, which causes this. If there's not IPython installed, `style='scvelo'` works just fine. The `ipywidgets` requirement from our side has already been removed on the `dev` branch in https://github.com/theislab/cellrank/pull/571.
